src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DockerImageDownloader.java

```diff
@@ -109,10 +111,25 @@ protected long download(long rangeStart, long rangeEnd, MessageDigest messageDig
     }
 
     @Override
-    public boolean downloadRequired() {
-        // TODO : Consider executing `docker image ls` to see if the required image version(tag/digest) already
-        //  exists to save a download attempt
-        return true;
+    public boolean downloadRequired() throws PackageDownloadException {
+        logger.atDebug().log("Checking if download is required");
+        checkDownloadPrerequisites();
+        Image image;
+        try {
+            image = Image.fromArtifactUri(artifact);
+        } catch (InvalidArtifactUriException e) {
+            logger.atError().log(String.format("Unexpected error making the image object from the provided artifact %s",
+                    artifact.getArtifactUri()));
+            throw new PackageDownloadException("Failed to check if the download required due to bad artifact URI", e);
+        }
+
+        if (image.getTag().equals(DOCKER_TAG_LATEST)) {
```

Comment:
flip this equals around: `DOCKER_TAG_LATEST.equals(image.getTag())` this protects from possible null pointers if getTag returned null.

Reply:
ACK